command/ca/provisionerbeta/update.go

```diff
@@ -0,0 +1,764 @@
+	if !ok {
+		return errors.New("error casting details to X5C type")
+	}
+	details := data.X5C
+	if ctx.IsSet("x5c-root") {
+		x5cRootFile := ctx.String("x5c-root")
+		roots, err := pemutil.ReadCertificateBundle(x5cRootFile)
+		if err != nil {
+			return errors.Wrapf(err, "error loading X5C Root certificates from %s", x5cRootFile)
+		}
+		var rootBytes [][]byte
+		for _, r := range roots {
+			if r.KeyUsage&x509.KeyUsageCertSign == 0 {
+				return errors.Errorf("error: certificate with common name '%s' cannot be "+
+					"used as an X5C root certificate.\n\n"+
+					"X5C provisioner root certificates must have the 'Certificate Sign' key "+
+					"usage extension.", r.Subject.CommonName)
+			}
+			rootBytes = append(rootBytes, pem.EncodeToMemory(&pem.Block{
+				Type:  "CERTIFICATE",
+				Bytes: r.Raw,
+			}))
+		}
+		details.Roots = rootBytes
+	}
+	return nil
+}
+
+func updateK8SSADetails(ctx *cli.Context, p *linkedca.Provisioner) error {
+	data, ok := p.Details.GetData().(*linkedca.ProvisionerDetails_K8SSA)
+	if !ok {
+		return errors.New("error casting details to K8SSA type")
+	}
+	details := data.K8SSA
+	if ctx.IsSet("public-key") {
+		pemKeysF := ctx.String("public-key")
+		pemKeysB, err := ioutil.ReadFile(pemKeysF)
+		if err != nil {
+			return errors.Wrap(err, "error reading pem keys")
+		}
+
+		var (
+			block   *pem.Block
+			rest    = pemKeysB
+			pemKeys = []interface{}{}
+		)
+		for rest != nil {
+			block, rest = pem.Decode(rest)
+			if block == nil {
+				break
+			}
+			key, err := pemutil.ParseKey(pem.EncodeToMemory(block))
+			if err != nil {
+				return errors.Wrapf(err, "error parsing public key from %s", pemKeysF)
+			}
+			switch q := key.(type) {
+			case *rsa.PublicKey, *ecdsa.PublicKey, ed25519.PublicKey:
+			default:
+				return errors.Errorf("Unexpected public key type %T in %s", q, pemKeysF)
+			}
+			pemKeys = append(pemKeys, key)
+		}
+
+		var pubKeyBytes [][]byte
+		for _, k := range pemKeys {
+			blk, err := pemutil.Serialize(k)
+			if err != nil {
+				return errors.Wrap(err, "error serializing pem key")
+			}
+			pubKeyBytes = append(pubKeyBytes, pem.EncodeToMemory(blk))
+		}
+		details.PublicKeys = pubKeyBytes
+	}
+	return nil
+}
+
+func updateOIDCDetails(ctx *cli.Context, p *linkedca.Provisioner) error {
+	data, ok := p.Details.GetData().(*linkedca.ProvisionerDetails_OIDC)
+	if !ok {
+		return errors.New("error casting details to OIDC type")
+	}
+	details := data.OIDC
+	if ctx.IsSet("client-id") {
+		details.ClientId = ctx.String("client-id")
+	}
+	if ctx.IsSet("client-secret") {
+		details.ClientSecret = ctx.String("client-secret")
+	}
+	if ctx.IsSet("remove-admin") {
+		details.Admins = removeElements(details.Admins, ctx.StringSlice("remove-admin"))
+	}
+	if ctx.IsSet("admin") {
+		details.Admins = append(details.Admins, ctx.StringSlice("admin")...)
+	}
+	if ctx.IsSet("domain") {
+		details.Domains = ctx.StringSlice("domain")
+	}
+	if ctx.IsSet("group") {
+		details.Groups = ctx.StringSlice("group")
+	}
```

Comment:
Just noting that if you want to change domains or groups you have to set all of them

Reply:
fixed. should have matched the others.